### PR TITLE
Metrics support pt2

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Plugging in metrics into the unleash client and added more tests around metrics corner cases

Followup to https://github.com/Unleash/unleash-proxy-client-swift/pull/36. We need to merge the previous PR first and this one will be around 100 LOC.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,10 +88,12 @@ This will stop and start the polling interval in order to renew polling with new
 
 ## Events
 
-The proxy client emits two different events you can subscribe to:
+The proxy client emits those different events you can subscribe to:
```
Comment (thomasheartman) [inline README.md:91]: Nice doc update! But "those different events" sounds a little off. How about something like this?

```suggestion
The proxy client emits events that you can subscribe to. The following events are available:
```

**Review by thomasheartman — APPROVED** [review]
This looks mostly good to me. I've got two non-blocking suggestion: one is a rewording suggestion, and the other has to do with the naming of the metrics events. I think it'd be worth addressing the one about metrics events naming before merging, at least, but I'll leave it at your discretion.

File: README.md [hunk 2]
```diff
@@ -88,10 +88,12 @@ This will stop and start the polling interval in order to renew polling with new
...
 
 - "ready"
 - "update"
+- "sent" (metrics sent)
+- "error" (metrics sending error)
```
[the remark this answers is not in the document]
Comment (kwasniew) [reply README.md:96]: "sent" and "error" are the same event names we have in Node.js client that I use as a reference impl. Currently error event is only emitted on metrics error since swift SDK didn't event emit any other error events before
